### PR TITLE
Provide the relevant types for the product.ts and update the appropriate docs

### PR DESCRIPTION
**Pull request by codyrancher**
https://github.com/rancher/dashboard/issues/8744 - `Fix the TS error regarding implicit any on the $plugin and store arguments`

### Note
The `Leaving these here for completeness but I don't think these should be advertised as usable to plugin creators.` comment sections should probably be removed. I just wanted the both of you to get a chance to comment if you think they should be a part of the interfaces or not.

### CODE REVIEW

File: docusaurus/docs/extensions/api/nav/custom-page.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ As we've seen from the previous chapter, a developer can register a top-level pr
 const BLANK_CLUSTER = '_';
 
 
-export function init($plugin, store) {
+export function init($plugin: IPlugin, store: any) {
```
Comment (aalves08) [inline docusaurus/docs/extensions/api/nav/custom-page.md:12]: I believe we need to `import { IPlugin } from '@shell/core/types'`. We should add it to all places, imho.

File: shell/types/rancher/index.d.ts [hunk 2]
```diff
@@ -5,3 +5,5 @@ declare module '@rancher/auto-import' {
 declare module '@shell/store/type-map' {
   export function DSL(store: any, name: string): any;
 }
+
+declare module '@shell/plugins/dashboard-store';
```
Comment (aalves08) [inline shell/types/rancher/index.d.ts:9]: Why do we need to declare this module?

**Review by aalves08 — COMMENTED** [review]
Added a couple of comments @codyrancher . Thanks for taking care of this mate! 🙏  Let me know if you need any help with it.

File: docusaurus/docs/extensions/api/nav/custom-page.md (hunk 1 above)
Comment (codyrancher) [reply docusaurus/docs/extensions/api/nav/custom-page.md:12]: Good catch. I fixed it.

**Review by jordojordo — COMMENTED** [review]
lgtm - just one minor comment about header options.

File: shell/core/types.ts [hunk 3]
```diff
@@ -136,6 +137,248 @@ export type LocationConfig = {
...
+   */
+  // ifHaveVerb: string | RegExp;
+  // removable: string;
+  // showWorkspaceSwitcher: boolean;
+  // supportRoute: string;
+  // to: string;
+  // typeStoreMap: string;
+}
+
+export interface HeaderOptions {
```
[the remark this answers is not in the document]
Comment (codyrancher) [reply shell/core/types.ts:227]: Added

**Review by jordojordo — APPROVED** [review]

**Pull request merged**